engines/registry/app/models/access_scope.rb

```diff
@@ -61,7 +61,10 @@ def authorized_actions(client = nil)
     if @namespace.nil?
       @image == 'catalog' ? @actions : AUTHORIZED_ACTION
     else
-      allowed_paths(client.systems.first)
+      @allowed_paths = []
+      allowed_paths(client.systems.first) if client.present?
```

Comment:
It's unexpected that the `allowed_paths` method internally sets the `@allowed_paths` variable. Also, the `allowed_paths` method has a default of `nil` for the system parameter, so why do we need to skip it when the client is nil? 

I'd suggest to write this as: 

`@allowed_paths = allowed_paths(client&.systems&.first)`

By skipping setting the allowed paths entirely when there is no client set, wouldn't that break accessing free images for anonymous clients?

Reply:
that was the first option I did, but inside of `allowed_paths` method we use the [system](https://github.com/SUSE/rmt/blob/master/engines/registry/app/models/access_scope.rb#L86) that would also break/raise exception

Reply:
happy to refactor the system use chain, just chose the smaller change to fix the issue

Reply:
If it breaks without a system, then the `system = nil` default is misleading. Are there no paths allowed for unauthenticated clients?

Reply:
yes, IIRC, the catalog check does a query without a system

Reply:
Done

Reply:
Done